# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../mod_fastcgi/lp381384.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

References:
http://thread.gmane.org/gmane.comp.web.fastcgi.devel/2613
http://article.gmane.org/gmane.comp.web.fastcgi.devel/1167
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=509116
https://bugs.launchpad.net/ubuntu/+source/libapache-mod-fastcgi/+bug/381384

Patch extracted from the SNAP-0811090952 snapshot tarball.
2009-05-28 Martin von Gagern

diff -ru mod_fastcgi-2.4.6/CHANGES mod_fastcgi-2.4.6-lp381384/CHANGES
--- mod_fastcgi-2.4.6/CHANGES	2007-11-13 13:26:35.000000000 +0100
+++ mod_fastcgi-2.4.6-lp381384/CHANGES	2009-05-28 21:02:31.000000000 +0200
@@ -1,3 +1,6 @@
+ *) Don't let the Content-Length header propagate on errors or across
+    redirects.  [Artur Zaprzala <arturz ownmail.net>]
+
 2.4.6
 
  *) Fix a bug I introduced in 2.4.4 that broke dynamic application
diff -ru mod_fastcgi-2.4.6/mod_fastcgi.c mod_fastcgi-2.4.6-lp381384/mod_fastcgi.c
--- mod_fastcgi-2.4.6/mod_fastcgi.c	2007-11-13 00:00:10.000000000 +0100
+++ mod_fastcgi-2.4.6-lp381384/mod_fastcgi.c	2009-05-28 21:02:31.000000000 +0200
@@ -753,6 +753,11 @@
                 ap_table_set(r->headers_out, "Location", value);
                 continue;
             }
+            
+            if (strcasecmp(name, "Content-Length") == 0) {
+                ap_table_set(r->headers_out, name, value);
+            	continue;
+            }
 
             /* If the script wants them merged, it can do it */
             ap_table_add(r->err_headers_out, name, value);
